### PR TITLE
380 fe edit additional information modal

### PR DESCRIPTION
**Pull request by aida-rodi**

### CODE REVIEW

**Review by jordimorillo — APPROVED** [review]

File: src/__tests__/hooks/useDragAndDropLanguagesHook.test.ts [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+  {
+    id: "d108e5aa-e058-4245-8a9a-e056a1594dfg",
+    name: "Anglais",
+    level: "Avançat",
+  }
+]
+
+describe("UseDragAndDropLanguagesHook", () => {
+
+  it("should render the App component", () => {
```
Comment (tomi-casabona) [inline src/__tests__/hooks/useDragAndDropLanguagesHook.test.ts:19]: The assertion name dont have relation with the test

File: src/__tests__/hooks/useDragAndDropLanguagesHook.test.ts [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+    act(() => {
+      result.current.handleDragStart(index)
+    })
+
+    expect(result.current.draggedItemIndex).toEqual(6)
+
+  })
+
+  it("should not update draggedItemIndex without calling handleDragStart", () => {
+    // Renderizamos el hook con los lenguajes
```
Comment (tomi-casabona) [inline src/__tests__/hooks/useDragAndDropLanguagesHook.test.ts:64]: delete comment

File: src/components/students/studentDetail/studentProfile/studentProfileCards/MyProfileLanguagesCard.tsx [hunk 3]
```diff
@@ -27,6 +32,10 @@ const MyProfileLanguagesCard: React.FC = () => {
             )}
             {!isLoadingLanguages && (
                 <div className="flex flex-col gap-2">
+                    <ModalPortals>
+                        {/* TODDO: Comentar a tomi por que aboutData.id es de typo numero */}
```
Comment (tomi-casabona) [inline src/components/students/studentDetail/studentProfile/studentProfileCards/MyProfileLanguagesCard.tsx:36]: No se porque estaba en tipo numero. se debe haber colado como error ! 
En mi rama ya lo he cambiado

Comment (tomi-casabona) [reply src/components/students/studentDetail/studentProfile/studentProfileCards/MyProfileLanguagesCard.tsx:36]: puedes quitar el c log

File: src/components/students/studentDetail/studentProfile/studentProfileCards/editStudentProfile/additionalInformation/DragAndDropLanguages.tsx [hunk 4]
```diff
@@ -0,0 +1,81 @@
+import { FC } from 'react'
+import {
+    TLanguage,
+    TLanguageLevel,
+} from '../../../../../../../interfaces/interfaces'
+// import { useUpdateLanguageHook } from '../../../../../../../hooks/useUpdateLanguageHook';
```
Comment (tomi-casabona) [inline src/components/students/studentDetail/studentProfile/studentProfileCards/editStudentProfile/additionalInformation/DragAndDropLanguages.tsx:6]: delete comment

File: src/components/students/studentDetail/studentProfile/studentProfileCards/editStudentProfile/additionalInformation/EditAdditionalInformation.tsx [hunk 5]
```diff
@@ -0,0 +1,249 @@
...
+
+    const {
+        updateLanguages,
+        availableLanguages,
+        deleteLanguage,
+        editLanguage,
+        
+    } = useUpdateLanguageHook(languagesData)
+
+    // TODDO: Refactor
```
Comment (tomi-casabona) [inline src/components/students/studentDetail/studentProfile/studentProfileCards/editStudentProfile/additionalInformation/EditAdditionalInformation.tsx:61]: delete comments

File: src/hooks/useDragAndDropLanguagesHook.ts [hunk 6]
```diff
@@ -0,0 +1,40 @@
...
+  const handleDragStart = useCallback((index: number) => {
+    setDraggedItemIndex(index);
+  }, []);
+
+  // Función para manejar el drag over (necesario para permitir el drop)
+  const handleDragOver = useCallback((e: React.DragEvent<HTMLDivElement>) => {
+    e.preventDefault();
+  }, []);
+
+  // Función para manejar el drop y reordenar la lista
```
Comment (tomi-casabona) [inline src/hooks/useDragAndDropLanguagesHook.ts:20]: delete comments

File: src/hooks/useStudentDetailHook.ts [hunk 7]
```diff
@@ -22,15 +22,21 @@ const useStudentDetailHook = (role?: string | null) => {
...
-            studentDetails.forEach((student) => {
-                dispatch(student(studentUUID))
-            })
-        }
+        const fetchData = async () => {
+            try {
+                const id = role === 'user' ? studentID : studentUUID;
+                if (!id) return; // Asegúrate de tener un ID válido
+
+                // Ejecuta todos los thunks en paralelo
```
Comment (tomi-casabona) [inline src/hooks/useStudentDetailHook.ts:30]: delete comments

**Review by tomi-casabona — APPROVED** [review]
Well done! 
I made few comments about deleting console.log and comments .

File: src/components/students/studentDetail/studentProfile/studentProfileCards/MyProfileLanguagesCard.tsx (hunk 3 above)
Comment (JuanDNJ) [reply src/components/students/studentDetail/studentProfile/studentProfileCards/MyProfileLanguagesCard.tsx:36]: jajaja

File: src/__tests__/hooks/useDragAndDropLanguagesHook.test.ts (hunk 1 above)
Comment (JuanDNJ) [reply src/__tests__/hooks/useDragAndDropLanguagesHook.test.ts:19]: Corregido

File: src/__tests__/hooks/useDragAndDropLanguagesHook.test.ts (hunk 2 above)
Comment (JuanDNJ) [reply src/__tests__/hooks/useDragAndDropLanguagesHook.test.ts:64]: Corregido

File: src/components/students/studentDetail/studentProfile/studentProfileCards/editStudentProfile/additionalInformation/DragAndDropLanguages.tsx (hunk 4 above)
Comment (JuanDNJ) [reply src/components/students/studentDetail/studentProfile/studentProfileCards/editStudentProfile/additionalInformation/DragAndDropLanguages.tsx:6]: Corregido

File: src/components/students/studentDetail/studentProfile/studentProfileCards/MyProfileLanguagesCard.tsx (hunk 3 above)
Comment (JuanDNJ) [reply src/components/students/studentDetail/studentProfile/studentProfileCards/MyProfileLanguagesCard.tsx:36]: Corregido

File: src/hooks/useStudentDetailHook.ts (hunk 7 above)
Comment (JuanDNJ) [reply src/hooks/useStudentDetailHook.ts:30]: Corregido

File: src/components/students/studentDetail/studentProfile/studentProfileCards/editStudentProfile/additionalInformation/EditAdditionalInformation.tsx (hunk 5 above)
Comment (JuanDNJ) [reply src/components/students/studentDetail/studentProfile/studentProfileCards/editStudentProfile/additionalInformation/EditAdditionalInformation.tsx:61]: Corregido

File: src/hooks/useDragAndDropLanguagesHook.ts (hunk 6 above)
Comment (JuanDNJ) [reply src/hooks/useDragAndDropLanguagesHook.ts:20]: Corregido

**Review by tomi-casabona — APPROVED** [review]

**Pull request merged**